openedx/core/djangoapps/programs/utils.py

```diff
@@ -127,6 +127,43 @@ def engaged_programs(self):
 
         return programs
 
+    @cached_property
+    def failed_course_runs(self):
+        """
+        Determine which course runs have been failed by the user.
+
+        Returns:
+            list of strings, each representing a certificate course run key
```

Comment:
how do we know where we use which. in catalog/utils and other places we refer to these as course runs.  
If we feel that course runs will be the more correct terminology moving forward, do we not want to gradually transition to calling them that?